clang/lib/AST/Expr.cpp

```diff
@@ -774,7 +774,15 @@ std::string PredefinedExpr::ComputeName(PredefinedIdentKind IK,
     const FunctionDecl *Decl = FD;
     if (const FunctionDecl* Pattern = FD->getTemplateInstantiationPattern())
       Decl = Pattern;
-    const FunctionType *AFT = Decl->getType()->getAs<FunctionType>();
+
+    // The type may not be fully initialized at this point.
+    // For example, in the trailing return type context of the lambda
+    // expression.
```

Comment:
```suggestion
  // Bail out of the type of the function has not been set yet.
  // This can notably happen in the trailing return type of a lambda expression
```